src/data/fintype/fiber.lean

```diff
@@ -0,0 +1,65 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Johan Commelin, Scott Morrison
+-/
+import data.fintype.basic
+
+/-!
+# The fibre over a point, when the domain is a fintype.
+
+We define the fiber of a function `f : α → β` over a point `b : β`, when we have `fintype α`,
+as a `finset α`, and prove some basic properties.
+-/
+
+open function
+
+namespace fintype
+/--
+The fiber of function `f : α → β` over a point `b : β` is a `finset α` when we have `fintype α`.
+-/
+def fiber {α : Type*} [fintype α] {β : Type*} [decidable_eq β] (f : α → β) (b : β) : finset α :=
+set.to_finset { a | f a = b }
+
+@[simp]
+lemma mem_fiber {α : Type*} [fintype α] {β : Type*} [decidable_eq β] (f : α → β) (b : β) (a : α) :
+  a ∈ fintype.fiber f b ↔ f a = b :=
+by simp [fiber]
+
+@[simp]
+lemma card_fiber {α : Type*} [fintype α] {β : Type*} [decidable_eq β] (f : α → β) (b : β) :
```

Comment:
Maybe we can also add `card_fiber_le` after this as extra helper lemma, stating that the card is less than `card \a`?

Reply:
Isn't this just [`fintype.card_subtype_le`](https://github.com/leanprover-community/mathlib/blob/4281343/src/data/fintype/basic.lean#L630)?